neqo-http3/src/connection_client.rs

```diff
@@ -2364,4 +2430,676 @@ mod tests {
         }
         assert!(recv_header && recv_data);
     }
+
+    #[allow(clippy::cognitive_complexity)]
+    fn check_control_qpack_request_streams_resumption(
+        mut server: &mut Connection,
+        expect_cap_instruction: bool,
+        expect_request: bool,
+    ) {
+        let mut control_stream = false;
+        let mut qpack_decoder_stream = false;
+        let mut qpack_encoder_stream = false;
+        let mut request = false;
+        while let Some(e) = server.next_event() {
+            match e {
+                ConnectionEvent::NewStream {
+                    stream_id,
+                    stream_type,
+                } => {
+                    assert!(
+                        (stream_id == 2)
+                            || (stream_id == 6)
+                            || (stream_id == 10)
+                            || (stream_id == 0)
+                    );
+                    if stream_id == 0 {
+                        assert_eq!(stream_type, StreamType::BiDi);
+                    } else {
+                        assert_eq!(stream_type, StreamType::UniDi);
+                    }
+                }
+                ConnectionEvent::RecvStreamReadable { stream_id } => {
+                    if stream_id == 2 {
+                        // the control stream
+                        // Only stream type will be received no Settings frame will be received.
+                        let mut buf = [0u8; 100];
+                        let (amount, fin) = server.stream_recv(stream_id, &mut buf).unwrap();
+                        assert_eq!(fin, false);
+                        assert_eq!(amount, CONTROL_STREAM_DATA.len());
+                        assert_eq!(&buf[..9], CONTROL_STREAM_DATA);
```

Comment:
This seems brittle.  When you define this, please define it alongside the default settings and explain their relationship.  That way, there is a better chance that they will get updated together.